codec/unknownproto/unknown_fields_test.go

```diff
@@ -661,89 +621,16 @@ func TestRejectUnknownFieldsFlat(t *testing.T) {
 	}
 }
 
-func TestMismatchedTypes_Nested(t *testing.T) {
```

Comment:
remove duplicate test cases

Reply:
Nit: I'd revert this so that we can focus on the actual fix given that the code removal of the unrelated section is more and thus distracting :)